src/audio/host.c

```diff
@@ -336,8 +337,9 @@ static uint32_t host_get_copy_bytes_normal(struct comp_dev *dev)
 
 	/* calculate minimum size to copy */
 	if (dev->direction == SOF_IPC_STREAM_PLAYBACK)
-		copy_bytes = MIN(avail_bytes,
-				 hd->local_buffer->stream.free);
+		copy_bytes = MIN(hd->period_bytes,
```

Comment:
ditto, just to explain the logic and reasoning.

Reply:
Done.